admin/worker/worker.go

```diff
@@ -94,6 +95,35 @@ func (w *Worker) schedule(ctx context.Context, name string, fn func(context.Cont
 	}
 }
 
+func (w *Worker) scheduleCron(ctx context.Context, name string, fn func(context.Context) error, cronExpr string) error {
+	loc, err := time.LoadLocation("America/Los_Angeles")
+	if err != nil {
+		return err
+	}
+
+	schedule, err := cron.ParseStandard(cronExpr)
+	if err != nil {
+		return err
+	}
+
+	for {
+		err := w.runJob(ctx, name, fn)
+		if err != nil {
+			return err
+		}
+
+		now := time.Now().In(loc)
+		nextRun := schedule.Next(now)
+		waitDuration := time.Until(nextRun)
+
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(waitDuration):
+		}
+	}
```

Comment:
I think you need to move the `runJob` call to *after* the `select`. Otherwise, the job would run every time the process restarts, which is not desired for cron-configured jobs.

Reply:
Oh, that's a good catch!  Just moved the logic based on you suggestion